peepingtom/depictors/napari/particledepictor.py

```diff
@@ -90,3 +98,65 @@ def update(self):
             for layer, vector in zip(self.vectors, vectors):
                 layer.data = vector.values
                 layer.scale = self.datablock.pixel_size
+
+
+class ParticleMeshDepictor(ParticleDepictor):
+    """
+    depict particles as isosurface meshes based on an input volume
+    volume: if not an ImageBlock or a xarray, axis order must be zyx
+    """
+    def __init__(self, datablock, *, volume, iso=None, step_size=1):
+        super().__init__(datablock)
+        self.volume = volume
+        self.iso = iso
+        self.step_size = step_size
+        self.mesh_color = 'cornflowerblue'
+
+    def depict(self):
+        vertices, faces = self.make_volume_mesh()
+        all_vertices, all_faces = self.tile_mesh(vertices, faces)
+
+        self._make_surface_layer(all_vertices,
+                                 all_faces,
+                                 name=f'{self.name} - particles',
+                                 scale=self.datablock.pixel_size,
+                                 # TODO: color
+                                 )
+
+    def make_volume_mesh(self):
+        vertices, faces, _, _ = marching_cubes(self.volume.copy(), level=self.iso, step_size=self.step_size)  # need copy for readonly issues
+        return vertices, faces
+
+    def tile_mesh(self, vertices, faces):
+        ori = self.datablock.orientations.data.values
+
+        # find rotated mesh vertices for each particle
+        all_vertices_rotated = (vertices @ ori)
```

Comment:
this reads strange mathematically, usually we apply a transform on a (column) vector by premultiplication
I think vertices should be reshaped to be column vectors (n, 3, 1) and this should be ori @ vertices

Reply:
I'm not entirely shure on this whole section of math... If you have suggestions like this, I'd love to see a PR with cleaner math :P